warehouse/warehouse.go

```diff
@@ -555,64 +664,262 @@ func (wh *HandleT) initWorkers() {
 				for {
 					// handle job to process staging files and convert them into load files
 					processStagingFilesJobList := <-wh.uploadToWarehouseQ
+					var whOneFullPassTimer *stats.RudderStats
+					for i, job := range processStagingFilesJobList {
+						if i == 0 {
+							whOneFullPassTimer = warehouseutils.DestStat(stats.TimerType, "total_end_to_end_step_time", job.Warehouse.Destination.ID)
```

Comment:
We call Start once (i = 0) but End for every job - Is this expected?

Reply:
modified